Src/Support/Google.Apis.Core/Http/HttpClientFactory.cs

```diff
@@ -88,18 +91,61 @@ protected virtual HttpMessageHandler CreateHandler(CreateHttpClientArgs args)
         /// <summary>
         /// Creates a simple client handler with redirection and compression disabled.
         /// </summary>
-        private HttpClientHandler CreateAndConfigureClientHandler()
+        private HttpMessageHandler CreateAndConfigureClientHandler()
         {
             var handler = CreateClientHandler();
-            if (handler.SupportsRedirectConfiguration)
+
+            ConfigureClientHandler(GetHttpClientHandler(handler));
+
+            return handler;
+        }
+
+        /// <summary>
+        /// Configure a simple client handler with redirection and compression disabled.
+        /// </summary>
+        private void ConfigureClientHandler(HttpClientHandler httpClientHandler)
+        {
+            if (httpClientHandler.SupportsRedirectConfiguration)
             {
-                handler.AllowAutoRedirect = false;
+                httpClientHandler.AllowAutoRedirect = false;
             }
-            if (handler.SupportsAutomaticDecompression)
+
+            if (httpClientHandler.SupportsAutomaticDecompression)
             {
-                handler.AutomaticDecompression = DecompressionMethods.None;
+                httpClientHandler.AutomaticDecompression = DecompressionMethods.None;
             }
-            return handler;
+        }
+
+        private HttpClientHandler GetHttpClientHandler(HttpMessageHandler httpMessageHandler)
```

Comment:
Or your `CustomCustomHttpClientFactory` could add this method and call it before returning on `CreateClientHandler`. It would be simpler even because you might know what is the real nature of the `HttpMessageHandler` that you are using.
Something like:

```
public class CustomHttpClientFactory<TService> : HttpClientFactory
    where TService : class
{
    private readonly IHttpMessageHandlerFactory _httpMessageHandlerFactory;

    public CustomHttpClientFactory(IHttpMessageHandlerFactory httpMessageHandlerFactory)
    {
        _httpMessageHandlerFactory = httpMessageHandlerFactory;
    }

    private HttpClientHandler GetHttpClientHandler(HttpMessageHandler httpMessageHandler)
    { ... }

    protected override HttpMessageHandler CreateClientHandler() => GetHttpClientHandler(_httpMessageHandlerFactory.CreateHandler(typeof(TService).Name));
}
```

Reply:
I can do that, but then I strip all middlewares (`DelegatingHandler`). E.g., for logging, ...

The point is to pass the whole pipeline into e.g. `StreamInterceptionHandler`

Reply:
So far, the only possibility to solve this in any other way is to open `StreamInterceptionHandler`, `GzipDeflateHandler` and `TwoWayDelegatingHandler` from internal to public.

Reply:
Yes, wrapping literaly with your method as it currently is will strip the whole pipeline if there's one, but you don't have to wrap exactly like that, that's what I meant with "you might know what is the real nature of the `HttpMessageHandler`" (and we necessarily don't for all `HttpMessageHandler` that might be returned from the factory). If you want to wrap with something very similar to your method that does not strip the pipeline, you can do as follows:
```
public class CustomHttpClientHandler : HttpClientHandler
{
    private HttpClientHandler _inner;
    private HttpMessageHandler _original;
    public static CustomHttpClientHandler FromHttpMessageHandler(HttpMessageHandler handler)
    {
          // Initialize a CustomHttpClientHandler setting _original to handler and
          // _inner to the value you currently extract in your proposed GetHttpClientHandler.
    }

    // Override relevant methods and properties, delegating on _inner or _original accordingly.
    // Not all properties are virtual, so, some copying of propety values from your CustomHttpClientHandler to _inner might be needed when overriding SendAsync.
}
public class CustomHttpClientFactory<TService> : HttpClientFactory
    where TService : class
{
    private readonly IHttpMessageHandlerFactory _httpMessageHandlerFactory;

    public CustomHttpClientFactory(IHttpMessageHandlerFactory httpMessageHandlerFactory)
    {
        _httpMessageHandlerFactory = httpMessageHandlerFactory;
    }

    protected override HttpMessageHandler CreateClientHandler() => CustomHttpClientHandler.FromHttpMessageHandler (_httpMessageHandlerFactory.CreateHandler(typeof(TService).Name));
}
```